calc.rb

```diff
@@ -0,0 +1,101 @@
+# Elsa's calculator project
+# Due: 150508
+# Breakfast + Lunch + Dinner
+
+# welcome message, instructions, and receiving user input
+print "\nWelcome to the Calculator!\n\nYou will be asked to enter\n  1) your first number,\n  2) your mathematical operation,\n  3) and your second number.\nThe resulting calculation will then be provided.\n"
+
+# sets up while loop for repeated calculations
+restart = "Y"
+while restart == "Y" do
+
+
+  # asks for the first number
+  print "\nFirst number:\t"
+  first_number = gets.chomp
+  # if not a number, asks again
+  while first_number != "0" && first_number.to_i == 0 do
+    puts "\nNice try. How about a number this time?"
+    print "First number:\t"
+    first_number = gets.chomp
+  end
+
+
+  # asks for mathematical operation (without immediate  validation *sad face*)
+  print "Math operation:\t"
+  operation = gets.chomp.downcase
+
+  # asks for a second number
+  print "Second number: \t"
+  second_number = gets.chomp
+  # if not a number, asks again
+  while second_number != "0" && second_number.to_i == 0 do
+    puts "\nNice try. How about a number this time?"
+    print "Second number:\t"
+    second_number = gets.chomp
+  end
+
+
+
+  # operator handling and result generator
+  case operation
```

Comment:
Fancy. Using `case` to get around having a long list of conditionals per if statement. :+1: